cmd/prometheus/main.go

```diff
@@ -46,6 +46,24 @@ func main() {
 	os.Exit(Main())
 }
 
+var (
+	configAttempts = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "prometheus",
+		Name:      "configuration_attempts",
+		Help:      "number of times a configuration reload was attempted.",
+	})
+	configSuccesses = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "prometheus",
+		Name:      "configuration_successes",
+		Help:      "number of times a configuration reload was successful.",
+	})
+	configSuccessTime = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "prometheus",
+		Name:      "configuration_last_success",
+		Help:      "Timestamp of the last successful configuration reload.",
+	})
+)
+
```

Comment:
Metric-naming- and doc-string-wise it'd be good to make this closer to our conventions at http://prometheus.io/docs/practices/naming/ (for example, counters should end with `_total`) and other code. I'd also include `reload` in the names to make it clearer what it's about. Something like this:

```
var (
	configAttempts = prometheus.NewCounter(prometheus.CounterOpts{
		Namespace: "prometheus",
		Name:      "config_reload_attempts_total",
		Help:      "Number of times a configuration reload was attempted.",
	})
	configSuccesses = prometheus.NewCounter(prometheus.CounterOpts{
		Namespace: "prometheus",
		Name:      "config_reload_successes_total",
		Help:      "Number of times a configuration reload was successful.",
	})
	configLastSuccessTime = prometheus.NewGauge(prometheus.GaugeOpts{
		Namespace: "prometheus",
		Name:      "config_last_reload_success_timestamp_seconds",
		Help:      "Unix timestamp in seconds of the last successful configuration reload.",
	})
)
```

When I think about what I'd want to monitor in terms of config reload failures, it's usually whether the last configuration reload was a successful one (which includes the one that always happens at startup). So maybe a boolean `1` / `0` gauge like `config_last_reload_successful` would make sense? Then you can alert if that gauge stays `0` for too long. @brian-brazil opinions?